### ISSUE TITLE
Report generation takes much more time with the new serenity 3.3.2 version

### ISSUE THREAD

**Issue by mrpet88** [serenity-bdd/serenity-core#2904]
We notice in my team, that after upgrading serenity the task aggregate (generating report) takes much more time with the new serenity version. In the screenshot you can compare old (2.5.8) and new (3.3.2) serenity build.

Some more info regarding our gradle file and version:

```

- 3.3.2 -> dependency classpath("net.serenity-bdd:serenity-gradle-plugin:3.3.2") instead of  classpath("net.serenity-bdd:serenity-gradle-plugin:2.5.8")
- 3.3.2 -> apply plugin: 'net.serenity-bdd.serenity-gradle-plugin' instead of 'net.serenity-bdd.aggregator'
- gradle 7.4.2 for both of the versions.

```

![Screenshot 2022-08-29 at 15 42 39](https://user-images.githubusercontent.com/40420772/187217490-911aacf7-b9e6-4ada-ae1d-6e2cbd14e0e3.png)
![Screenshot 2022-08-29 at 15 41 49](https://user-images.githubusercontent.com/40420772/187217493-b0830267-bf4d-400c-aae7-c41062ff1677.png)

**Comment by wakaleo** [comment]
It is what it is. Serenity 3.x reports are much more comprehensive than Serenity 2.x reports. Without more detailed info (timings, memory analysis, sample projects) there isn't a lot we can do.

**Comment by wakaleo** [comment]
Fixed in 3.5.0.
